libbeat/outputs/logstash/config.go

```diff
@@ -11,6 +11,16 @@ type logstashConfig struct {
 	CompressionLevel int                `config:"compression_level"`
 	MaxRetries       int                `config:"max_retries"`
 	TLS              *outputs.TLSConfig `config:"tls"`
+	SOCKSProxy       *socksProxyConfig  `config:"socks_proxy"`
+}
+
+// socksProxyConfig holds the configuration information required to proxy
+// Logstash connections through a SOCKS5 server.
+type socksProxyConfig struct {
+	HostPort  string `config:"hostport"`   // host:port of the SOCKS5 server.
```

Comment:
I updated the PR. The config is now consistent with Elasticsearch (but requires a `socks5://` URL).

```
output:
  logstash:
    hosts: ["remote-host:5044"]
    proxy_url: socks5://user:password@socks5-proxy:2233
```